### PR TITLE
feat(text & heading): add Text and Heading Component

### LINKED ISSUE #30: [RFC] Heading and Paragraph Components
## Intro
image
We will be creating the Text component for body1, body2 & body3 as p tag and Heading component for Headings as h1, h2.

## The Goal:
Create `<Text />`,` <Text.Body1 />`, `<Text.Body2 />`, `<Text.Body3 />` for Paragraphs.
Create `<Heading>`, `<Heading.H1 />`, `<Heading.H2 />` for Headings.

## Key Features:
* Override Support

  * Typography
  * Space
  * Color

* Dot Notation styling using `variant` prop
 
## Export:

**Component Parts**

* `Text`: Base component using Chakra `Text` Component but restricted to props with override support.
* `Text.Body1:` Body 1 variant for our app styled only by `theme.js`.
* `Text.Body2`: Body 2 variant for our app styled only by `theme.js`.
* `Text.Body3`: Body 2 variant for our app styled only by `theme.js`.
* `Heading`:  Base component using Chakra `Box` Component as h1 but restricted to props with override support
* `Heading.H1`:  Heading 1 variant for our app styled only by `theme.js`.
* `Heading.H2`:  Heading 2 variant for our app styled only by `theme.js`.

## Usage

**Heading**

```js-jsx
<Heading>No Styling</Heading>
<Heading.H1>Heading 1</Heading.H1>
<Heading.H2>Heading 2</Heading.H2>
```

**Paragraph**

```js-jsx
<Text>No Styling</Text>
<Text.Body1>Body 1</Text.Body1>
<Text.Body2>Body 2</Text.Body2>
<Text.Body3>Body 3</Text.Body3>
```

## Props

* `variant` - For styling the components based on base component only.
* `override` - For supporting override from external but should be used only in case of necessity.
* `children` - This should be only text.

## `theme.js`:

```js
import React from 'react';
import { theme } from '@chakra-ui/core';

export default {
  ...theme,
  lineHeights: {
    ...theme.lineHeights,
    h1: '24px',
    h2: '20px',
    body1: '19px',
    body2: '17px',
    body3: '16px',
  },
  colors: {
    ...theme.colors,
    font: {
      heading: '#282929',
      body1: '#8C8D8E',
      body2: '#5D5E5E',
      body3: '#282929',
    },
  },
  fontSizes: {
    ...theme.fontSizes,
    h1: '20px',
    h2: '16px',
    body1: '15px',
    body2: '14px',
    body3: '13px',
  },
  letterSpacings: {
    h1: '0.2px',
    h2: '0.18px',
    body1: '-0.15px',
    body2: '-0.14px',
  },
  fonts: {
    ...theme.fonts,
    heading: `Inter,${theme.fonts.heading}`,
    body: `Inter,${theme.fonts.body}`,
  },
};

```

### PR DESCRIPTION
**Pull request by navin-moorthy**
Create <Text />, <Text.Body1 />, <Text.Body2 />, <Text.Body3 /> for Paragraphs.Create
<Heading>,<Heading.H1 />, <Heading.H2 /> for Headings.

[RFC] Heading and Paragraph Components #30

### CODE REVIEW

**Comment by navin-moorthy** [comment]
Base Component are added in /src.
I will be adding the docz now.

File: docs/packages/components/Checkbox/index.js [hunk 1]
```diff
@@ -26,17 +33,23 @@ export const Checkbox = ({
...
   const Override = {
     wrapper: {
+      textAlign: 'center',
       lineHeight: 'checkbox',
+      display: 'inline-flex',
+      flexDirection: isReversed ? 'row-reverse' : 'row',
+      justifyContent: isReversed ? 'flex-end' : 'start',
       cursor: isDisabled ? 'not-allowed' : 'pointer',
-      ...getAllowedProps(override.wrapper, ['typography']),
+      ...getAllowedProps(override.wrapper, [...layout, ...typography, ...flex]),
```
Comment (prasanna1211) [inline docs/packages/components/Checkbox/index.js:46]: I don't think layout is required here. Should wrap this with a box.

File: docs/packages/components/Checkbox/index.js [hunk 2]
```diff
@@ -50,60 +63,63 @@ export const Checkbox = ({
...
+            {...Override.visuallyHidden}
+          />
 
-            {/* This is the control box with a check icon as children */}
-            <ControlBox {...Override.controlBox}>
-              <Icon name="customCheck" size="10px" />
-            </ControlBox>
+          {/* This is the control box with a check icon as children */}
+          <ControlBox {...Override.controlBox}>
+            <Icon name="customCheck" size="10px" />
```
Comment (prasanna1211) [inline docs/packages/components/Checkbox/index.js:114]: Hasn't this been custom icon ?

File: docs/packages/components/Checkbox/index.js (hunk 1 above)
Comment (navin-moorthy) [reply docs/packages/components/Checkbox/index.js:46]: These changes where done by Govind, I haven't looked very closely on the props. My concentration was to bring the docz up properly for all component.
But here Layout is not required for checkbox.
But we need the space

Comment (prasanna1211) [reply docs/packages/components/Checkbox/index.js:46]: @navin-moorthy The build for this branch breaks. Is it this branch where the docz are working for Text and Heading component ?

**Review by prasanna1211 — APPROVED** [review]
Merging this without reviewing. Going to change the structure in development. Few learnings that can be taken

1. Keep PRs small. For example don't have folder restructure and components in same branch
2. Update readme for starting the app.

**Pull request merged**